plugin/src/main/java/com/h3xstream/findsecbugs/crypto/InsufficientKeySizeRsaDetector.java

```diff
@@ -118,22 +116,21 @@ private void analyzeMethod(Method m, ClassContext classContext) throws CFGBuilde
                         }
                     }
 
-                    if (n != null && n.intValue() < 1024) {
-                        initializeWeakKeyLength = true;
-                        locationWeakness = location;
+                    if (n != null && n.intValue() < 2048 && createRsaKeyGen) {
+                        addToReport(m, classContext, location, n);
                     }
                 }
             }
-        }
+        }       
+    }
 
-        //Both condition have been found in the same method
-        if (createRsaKeyGen && initializeWeakKeyLength) {
-            JavaClass clz = classContext.getJavaClass();
-            bugReporter.reportBug(new BugInstance(this, RSA_KEY_SIZE_TYPE, Priorities.NORMAL_PRIORITY) //
-                    .addClass(clz)
-                    .addMethod(clz, m)
-                    .addSourceLine(classContext, m, locationWeakness));
-        }
+    private void addToReport(Method m, ClassContext classContext, Location locationWeakness, Number n){
+        JavaClass clz = classContext.getJavaClass();
+        int priority = (n.intValue() < 1024) ? Priorities.NORMAL_PRIORITY : Priorities.LOW_PRIORITY;
```

Comment:
Excellent. **Priority** stands for **confidence** in FindBugs and this reflect perfectly the exploitability of each key size.